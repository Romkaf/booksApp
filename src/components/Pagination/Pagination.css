.root {
	display: flex;
	flex-wrap: wrap;
	align-content: center;
	margin: 20px 0;
	padding: 0;
	list-style: none;
}

.page {
	border: 1px solid var(--grey);
	border-radius: 100%;
	width: 43px;
	height: 43px;
	margin: 0 10px;
	margin-bottom: 10px;
	cursor: pointer;
	transition: linear 0.2s;
}

.pageCurrent {
	border: 2px solid var(--semi-black);
	font-weight: 700;
	cursor: auto;
	pointer-events: none;
}

.page:hover:not(.pageCurrent),
.page:focus {
	box-shadow: 0 0 5pt 1.5pt var(--grey);
	outline: none;
}
