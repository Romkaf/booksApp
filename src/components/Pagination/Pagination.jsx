import React from 'react';
import PropTypes from 'prop-types';
import styles from './Pagination.css';

const Pagination = ({ pages = [], onPageClick, currentPage }) => {
	const { root, page, pageCurrent } = styles;

	return (
		<ul className={root}>
			{pages.map((it, idx) => (
				<li key={idx}>
					<button
						className={currentPage == it ? `${page} ${pageCurrent}` : `${page}`}
						aria-label="pagination-page"
						onClick={onPageClick(it)}
						tabIndex={currentPage == it ? -1 : 0}
					>
						{it}
					</button>
				</li>
			))}
		</ul>
	);
};

Pagination.propTypes = {
	pages: PropTypes.arrayOf(PropTypes.number),
	onPageClick: PropTypes.func,
	currentPage: PropTypes.number,
};

export default Pagination;
