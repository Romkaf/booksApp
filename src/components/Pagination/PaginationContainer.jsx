import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { createPages } from '@utils/createPages';
import { setCurrentPage } from '@models/actions';
import Pagination from './Pagination';

const PaginationContainer = () => {
	const dispatch = useDispatch();

	const { currentPage, perPage } = useSelector((state) => state.pagination);
	const totalCount = useSelector((state) => state.books.totalCount);

	const handlePageClick = (page) => () => dispatch(setCurrentPage(page));

	const pagesCount = Math.ceil(totalCount / perPage);
	const pages = createPages(pagesCount, currentPage);

	return (
		<Pagination
			pages={pages}
			onPageClick={handlePageClick}
			currentPage={currentPage}
		/>
	);
};

export default PaginationContainer;
