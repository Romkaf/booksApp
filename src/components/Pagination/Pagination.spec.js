import React from 'react';
import Pagination from './Pagination';
import { shallow } from 'enzyme';

describe('Pagination', () => {
	const props = {
		pages: [1, 2, 3],
		onPageClick: (it) => () => it,
		currentPage: 1,
	};

	let component;
	const setUp = (props) => shallow(<Pagination {...props} />);

	describe('should render component', () => {
		it('with props', () => {
			component = setUp(props);
			expect(component).toMatchSnapshot();
		});

		it('without props', () => {
			component = setUp();
			expect(component).toMatchSnapshot();
		});
	});

	// it('should call handler when button will click', () => {
	// 	const mockFn = jest.fn((it) => it);
	// 	component = setUp(props);
	// 	component.setProps({ onPageClick: (it) => () => mockFn(it) });
	// 	const firstButton = component.find('ul').childAt(0).find('button');
	// 	firstButton.simulate('click');
	// 	expect(mockFn).toHaveReturnedWith(1);
	// });
});
