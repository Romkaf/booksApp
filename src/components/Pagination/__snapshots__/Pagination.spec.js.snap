// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render component with props 1`] = `
<ul
  className="root"
>
  <li
    key="0"
  >
    <button
      aria-label="pagination-page"
      className="page pageCurrent"
      onClick={[Function]}
      tabIndex={-1}
    >
      1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      aria-label="pagination-page"
      className="page"
      onClick={[Function]}
      tabIndex={0}
    >
      2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      aria-label="pagination-page"
      className="page"
      onClick={[Function]}
      tabIndex={0}
    >
      3
    </button>
  </li>
</ul>
`;

exports[`Pagination should render component without props 1`] = `
<ul
  className="root"
/>
`;
